failed: typed.RecoveryPermitterSpec

https://jenkins.akka.io:8498/job/akka-multi-node-nightly/6099

```
[ERROR] [06/15/2018 05:13:04.026] [RecoveryPermitterSpec-akka.actor.default-dispatcher-5] [akka://RecoveryPermitterSpec/user/$e/$a][persistenceId:p3][phase:replay-evts] Exception during recovery.  Last known sequence number [p3] WARNING arguments left: 1 (akka.persistence.typed.internal.RecoveryPermitterSpec$TE: Boom!)
[ERROR] [06/15/2018 05:13:04.032] [RecoveryPermitterSpec-akka.actor.default-dispatcher-6] [akka://RecoveryPermitterSpec/user/$e] death pact with Actor[akka://RecoveryPermitterSpec/user/$e/$a#-1786459255] was triggered
akka.actor.typed.DeathPactException: death pact with Actor[akka://RecoveryPermitterSpec/user/$e/$a#-1786459255] was triggered
	at akka.actor.typed.internal.adapter.ActorAdapter.unhandled(ActorAdapter.scala:120)
	at akka.actor.typed.internal.adapter.ActorAdapter.akka$actor$typed$internal$adapter$ActorAdapter$$next(ActorAdapter.scala:71)
	at akka.actor.typed.internal.adapter.ActorAdapter$$anonfun$running$1.applyOrElse(ActorAdapter.scala:48)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:517)
	at akka.actor.typed.internal.adapter.ActorAdapter.aroundReceive(ActorAdapter.scala:22)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:588)
	at akka.actor.dungeon.DeathWatch$class.receivedTerminated(DeathWatch.scala:67)
	at akka.actor.ActorCell.receivedTerminated(ActorCell.scala:431)
	at akka.actor.ActorCell.autoReceiveMessage(ActorCell.scala:573)
	at akka.actor.ActorCell.invoke(ActorCell.scala:556)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
	at akka.dispatch.Mailbox.run(Mailbox.scala:225)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

[INFO] [06/15/2018 05:13:04.032] [RecoveryPermitterSpec-akka.actor.default-dispatcher-5] [akka://RecoveryPermitterSpec/user/$e] Message [akka.persistence.typed.internal.RecoveryPermitterSpec$StopActor$] without sender to Actor[akka://RecoveryPermitterSpec/user/$e#2125709900] was not delivered. [1] dead letters encountered. If this is not an expected behavior, then [Actor[akka://RecoveryPermitterSpec/user/$e#2125709900]] may have terminated unexpectedly, This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[info] - must return permit when actor throws from RecoveryCompleted *** FAILED *** (3 seconds, 12 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMessageClass waiting for interface akka.actor.typed.ActorRef
[info]   at scala.Predef$.assert(Predef.scala:170)
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessageClass_internal(TestProbeImpl.scala:178)
[info]   at akka.actor.testkit.typed.scaladsl.TestProbe$class.expectMessageType(TestProbe.scala:146)
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessageType(TestProbeImpl.scala:45)
[info]   at akka.persistence.typed.internal.RecoveryPermitterSpec$$anonfun$1$$anonfun$apply$mcV$sp$6.apply$mcV$sp(RecoveryPermitterSpec.scala:205)
[info]   at akka.persistence.typed.internal.RecoveryPermitterSpec$$anonfun$1$$anonfun$apply$mcV$sp$6.apply(RecoveryPermitterSpec.scala:181)
[info]   at akka.persistence.typed.internal.RecoveryPermitterSpec$$anonfun$1$$anonfun$apply$mcV$sp$6.apply(RecoveryPermitterSpec.scala:181)
```

--- a/akka-persistence-typed/src/test/scala/akka/persistence/typed/internal/RecoveryPermitterSpec.scala
+++ b/akka-persistence-typed/src/test/scala/akka/persistence/typed/internal/RecoveryPermitterSpec.scala
@@ -5,6 +5,7 @@
 package akka.persistence.typed.internal
 
 import akka.actor.PoisonPill
+import akka.actor.testkit.typed.scaladsl.{ ActorTestKit, TestProbe }
 import akka.actor.typed.scaladsl.Behaviors
 import akka.actor.typed.scaladsl.adapter.{ TypedActorRefOps, TypedActorSystemOps }
 import akka.actor.typed.{ ActorRef, Behavior, TypedAkkaSpecWithShutdown }
@@ -12,7 +13,7 @@ import akka.persistence.Persistence
 import akka.persistence.RecoveryPermitter.{ RecoveryPermitGranted, RequestRecoveryPermit, ReturnRecoveryPermit }
 import akka.persistence.typed.scaladsl.PersistentBehaviors.CommandHandler
 import akka.persistence.typed.scaladsl.{ Effect, PersistentBehaviors }
-import akka.actor.testkit.typed.scaladsl.{ ActorTestKit, TestProbe }
+import akka.testkit.EventFilter
 import com.typesafe.config.{ Config, ConfigFactory }
 import org.scalatest.concurrent.Eventually
 
@@ -71,9 +72,11 @@ class RecoveryPermitterSpec extends ActorTestKit with TypedAkkaSpecWithShutdown 
         akka.persistence.max-concurrent-recoveries = 3
         akka.persistence.journal.plugin = "akka.persistence.journal.inmem"
         akka.actor.warn-about-java-serializer-usage = off
+        akka.loggers = ["akka.testkit.TestEventListener"]
       """)
+  implicit val untypedSystem = system.toUntyped
 
-  private val permitter = Persistence(system.toUntyped).recoveryPermitter
+  private val permitter = Persistence(untypedSystem).recoveryPermitter
 
   def requestPermit(p: TestProbe[Any]): Unit = {
     permitter.tell(RequestRecoveryPermit, p.ref.toUntyped)
@@ -183,22 +186,24 @@ class RecoveryPermitterSpec extends ActorTestKit with TypedAkkaSpecWithShutdown 
       requestPermit(p2)
 
       val stopProbe = TestProbe[ActorRef[Command]]()
-      val parent = spawn(
-        Behaviors.setup[Command](ctx ⇒ {
-          val persistentActor =
-            ctx.spawnAnonymous(persistentBehavior("p3", p3, p3, throwOnRecovery = true))
-          ctx.watch(persistentActor)
-          Behaviors.receive[Command] {
-            case (_, StopActor) ⇒
-              stopProbe.ref ! persistentActor
-              ctx.stop(persistentActor)
-              Behavior.same
-            case (_, message) ⇒
-              persistentActor ! message
-              Behaviors.same
-          }
-        })
-      )
+      val parent =
+        EventFilter.error(occurrences = 1, start = "Exception during recovery.").intercept {
+          spawn(
+            Behaviors.setup[Command](ctx ⇒ {
+              val persistentActor =
+                ctx.spawnAnonymous(persistentBehavior("p3", p3, p3, throwOnRecovery = true))
+              Behaviors.receive[Command] {
+                case (_, StopActor) ⇒
+                  stopProbe.ref ! persistentActor
+                  ctx.stop(persistentActor)
+                  Behavior.same
+                case (_, message) ⇒
+                  persistentActor ! message
+                  Behaviors.same
+              }
+            })
+          )
+        }
       p3.expectMessage(Recovered)
       // stop it
       parent ! StopActor

